pkg/apis/dashboard/migration/schemaversion/v36.go

```diff
@@ -0,0 +1,137 @@
+package schemaversion
+
+func V36(dsInfo DataSourceInfoProvider) SchemaVersionMigrationFunc {
+	datasources := dsInfo.GetDataSourceInfo()
+	return func(dashboard map[string]interface{}) error {
+		dashboard["schemaVersion"] = int(36)
+
+		// getDataSourceRef creates a standardized ref object from a datasource
+		getDataSourceRef := func(ds map[string]interface{}) map[string]interface{} {
+			ref := map[string]interface{}{
+				"uid":  ds["uid"],
+				"type": ds["type"],
+			}
+			if apiVersion, ok := ds["apiVersion"]; ok {
+				ref["apiVersion"] = apiVersion
+			}
+			return ref
+		}
+
+		// Mock for getInstanceSettings since we don't have access to datasource service
+		getInstanceSettings := func(nameOrRef interface{}) map[string]interface{} {
+			for _, _ = range datasources {
+				// TODO: Implement logic to find datasource by nameOrRef
```

Comment:
@ivanortegaalba this is where you'd implement the logic needed for filtering the list of datasources

Reply:
I just created this one to get a DS by UID. But it is up to change it. It was just a way to organize myself :)